scripts/build_qat_envoy.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+if [ -f /etc/os-release ]; then
+	source /etc/os-release
+else
+	source /usr/lib/os-release
+fi
+
+readonly YQ_BIN="$(realpath ${PWD}/yq)"
+readonly BAZEL_BIN="$(realpath ${PWD}/bazel.sh)"
+readonly SCRIPTS_DIR="$(realpath "$(dirname "$0")")"
+readonly VERSIONS_FILE="${SCRIPTS_DIR}/../versions.yaml"
+readonly OPENSSL_DIR="${SCRIPTS_DIR}/../openssl"
+readonly QAT_ENGINE_DIR="${SCRIPTS_DIR}/../QAT_Engine"
+readonly ENVOY_DIR="${SCRIPTS_DIR}/../envoy"
+readonly QAT_LIB_DIR="${SCRIPTS_DIR}/../QAT_Lib"
+
+source "${SCRIPTS_DIR}/lib.sh"
+
+install_yq() {
+	info "Download and install yq"
+	local yq_arch=""
+	case $(uname -m) in
+		x86_64)
+			yq_arch=amd64
+		;;
+
+		*)
+			die "Unsupported architecture: $(uname -m)"
+	esac
+
+	curl -L https://github.com/mikefarah/yq/releases/download/2.4.0/yq_linux_${yq_arch} -o ${YQ_BIN}
+	chmod +x ${YQ_BIN}
+}
+
+install_bazel() {
+	info "Download and install bazel"
+	local bazel_url="$(${YQ_BIN} r "${VERSIONS_FILE}" "tools.bazel.url")"
+	local bazel_version="$(${YQ_BIN} r "${VERSIONS_FILE}" "tools.bazel.version")"
```

Comment:
I think it's a good idea to make the versions parametric. Different Envoy versions require different bazel versions to build. You could add the required bazel checksums to the configuration file to verify the downloads.

Reply:
sure thing!